### PR TITLE
Clean up PHP parser

### PR DESCRIPTION
**Pull request by AJenbo**
Type: refactoring
Breaking change: no

To keep the size of the PR down test have not been merged, this will happen in a follow up once merged.

### CODE REVIEW

**Review by kylekatarnls — DISMISSED** [review]

File: src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php [hunk 1]
```diff
@@ -7388,6 +7450,7 @@ protected function isScalarOrCallableTypeHint($image)
             case 'callable':
             case 'iterable':
             case 'void':
+            case 'never':
                 return true;
         }
```
[the remark this answers is not in the document]
Comment (ravage84) [reply src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:7455]: Agreed but can be done later.

[the remark this answers is not in the document]
Comment (AJenbo) [reply src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:7455]: By anyone :D

**Review by ravage84 — APPROVED** [review]
I'm trusting @kylekatarnls' judgement

**Comment by AJenbo** [comment]
Nice, this unblocked most of the other PRs.

**Pull request merged**